app/controllers/classrooms_controller.rb

```diff
@@ -61,4 +65,20 @@ def set_classroom
   def classroom_params
     params.require(:classroom).permit(:name, :grade, :school_name, :year_value)
   end
+
+  def ensure_teacher_or_admin
```

Comment:
No required, but you might want to consider making this a controller concern since you have it in more than one. Or, adding it to ApplicationController